src/lib/incentive-relationship-calculation.ts

```diff
@@ -93,25 +122,50 @@ export function buildRelationshipGraph(data: IncentiveRelationships) {
   return edges;
 }
 
-// Checks the prerequisites for a single incentive.
-export function meetsPrerequisites(
+// Helper to traverse nested prerequisites and check whether they are met for
+// the given incentive.
+function meetsSinglePrerequisite(
   incentiveId: string,
+  prerequisite: IncentivePrerequisites,
   maps: RelationshipMaps,
 ) {
-  const prerequisiteIds = maps.requiresMap.get(incentiveId);
-  if (prerequisiteIds !== undefined) {
-    for (const prerequisiteId of prerequisiteIds) {
-      if (
-        !maps.eligibleIncentives.has(prerequisiteId) &&
-        maps.eligibleIncentives.has(incentiveId)
-      ) {
+  if (typeof prerequisite === 'string') {
+    if (
+      !maps.eligibleIncentives.has(prerequisite) &&
+      maps.eligibleIncentives.has(incentiveId)
```

Comment:
I prefer this as a guard clause at the top since you can deal with this case without knowing anything else. If the base incentive is already ineligible, we don't need to do any more checking.

Same is true with the IsExcluded method below (but I can't comment on that since it's not in the diff :)

Reply:
We have to check that `prerequisite` is a string before checking for it in the map, so for this one it has to stay like this (I can fix it in IsExcluded though).